model/configuration.py

```diff
@@ -769,3 +774,567 @@ def excluded_audio_data_sources(cls, _db):
+        """Initializes a new instance of ConfigurationBucket
+
+        :param configuration_storage: ConfigurationStorage object
+        :type configuration_storage: BaseConfigurationStorage
+
+        :param db: Database session
+        :type db: sqlalchemy.orm.session.Session
+        """
+        self._logger = logging.getLogger()
+        self._configuration_storage = configuration_storage
+        self._db = db
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self._db = None
+
+    def get_setting_value(self, setting_name):
+        """Returns a settings'value
+
+        :param setting_name: Name of the setting
+        :type setting_name: string
+
+        :return: Setting's value
+        :rtype: Any
+        """
+        return self._configuration_storage.load(self._db, setting_name)
+
+    def set_setting_value(self, setting_name, setting_value):
+        """Sets setting's value
+
+        :param setting_name: Name of the setting
+        :type setting_name: string
+
+        :param setting_value: New value of the setting
+        :type setting_value: Any
+        """
+        self._configuration_storage.save(self._db, setting_name, setting_value)
+
+    @classmethod
+    def to_settings(cls):
+        settings = []
+
+        for name, member in ConfigurationMetadata.get_configuration_metadata(cls):
+            key_attribute = getattr(member, ConfigurationAttribute.KEY.value, None)
+            label_attribute = getattr(member, ConfigurationAttribute.LABEL.value, None)
+            description_attribute = getattr(member, ConfigurationAttribute.DESCRIPTION.value, None)
+            type_attribute = getattr(member, ConfigurationAttribute.TYPE.value, None)
+            required_attribute = getattr(member, ConfigurationAttribute.REQUIRED.value, None)
+            default_attribute = getattr(member, ConfigurationAttribute.DEFAULT.value, None)
+            options_attribute = getattr(member, ConfigurationAttribute.OPTIONS.value, None)
+            category_attribute = getattr(member, ConfigurationAttribute.CATEGORY.value, None)
+
+            settings.append({
+                ConfigurationAttribute.KEY.value: key_attribute,
+                ConfigurationAttribute.LABEL.value: label_attribute,
+                ConfigurationAttribute.DESCRIPTION.value: description_attribute,
+                ConfigurationAttribute.TYPE.value: type_attribute.to_control_type(),
+                ConfigurationAttribute.REQUIRED.value: required_attribute,
+                ConfigurationAttribute.DEFAULT.value: default_attribute,
+                ConfigurationAttribute.OPTIONS.value:
+                    [option.to_settings() for option in options_attribute]
+                    if options_attribute
+                    else None,
+                ConfigurationAttribute.CATEGORY.value: category_attribute
+            })
+
+        return settings
+
+
+class ConfigurationFactory(object):
```

Comment:
I'm not sure about this; note to myself to come back once I see how it's used.

Reply:
Here is a small example of how it's used:
```
with self._configuration_factory.create(
                self._configuration_storage, db, LCPEncryptionConfiguration) as configuration:

            if self._lcpencrypt_exists_locally(configuration):
                result = self._run_lcpencrypt_locally(file_path, identifier, configuration)

                return result
```

The general idea is to move instantiation of `ConfigurationBucket` descendants out of the client code. `ConfigurationBucket` descendants have not to be created directly because `ConfigurationBucket`'s constructor accepts a database session and storing it may lead to the problems (like accessing session aware objects using a stale session). Instead `ConfigurationFactory` abstracts this process and provides a way to instantiate `ConfigurationBucket` descendants using a context manager

Reply:
OK, I don't think this is how I'd do it but it makes sense.